### DISCUSSION TITLE
Fucntion to detect blur in scanned text pdf pages

### DISCUSSION THREAD

**Discussion by ghareesh** [ImageMagick/ImageMagick#2185] [Help]
Hello ..
I am currently developing automated content extraction solution for reading contracts.. 
Some documents scanned are blurry.. Is there an inbuilt function in Imagemagick to detect blurriness in an image, so that I can apply deblurring selectively to those images..

**Comment by snibgo** [comment]
There are essentially two methods.

The first is purely numeric, eg the maximum standard over a sliding window, eg 5x5:
```
magick in.png -statistic standard_deviation 5x5 -format %[fx:maxima] info:

0.196643
```
Look at this number for inputs that you consider "sharp" and "blurred". This will give you a threshold value.

The second method is to sharpen the image and find how much it has changed. If it hasn't changed much, it was already sharp.

See also my page [Measuring blur](http://im.snibgo.com/measblur.htm) which gives a more complex algorithm, an elaboration of the second method.

**Reply by ghareesh** [reply]
Its heavy.. I will try out.. Thanks @snibgo !!
